src/matlab2tikz.m

```diff
@@ -3606,6 +3690,31 @@ function warnFacetedInterp(m2t)
                        env, drawOpts, tabOpts, table);
 end
 % ==============================================================================
+function dataInfo = getDataInfo(h, varargin)
+    % retrieves the "*Data  fields from a HG object
+    % When no names are specified, it assumes 'X','Y','Z' is requested
+    if nargin == 1
+        fields = {'X','Y','Z'};
+    else
+        fields = varargin;
+    end
+    dataInfo = struct();
+    for iField = 1:numel(fields)
+        name            = fields{iField};
+        dataInfo.(name) = get(h, [name 'Data']);
+    end
+end
+% ==============================================================================
+function value = tryToMakeScalar(value, tolerance)
+    % make a vector into a scalar when all its components are equal
+    if ~exist('tolerance','var')
+        tolerance = 0; % do everything perfectly
+    end
+    if all(abs(value - value(1)) <= tolerance)
```

Comment:
There is a maltab function "uniquetol". If that also exists for octave we could use it here.